### DISCUSSION TITLE
Does TipTap sanitize HTML or do I have to use a dedicated package?

### DISCUSSION THREAD

**Discussion by Arturexe** [ueberdosis/tiptap#2845] [General Questions]
Before working with TipTap I was using `DOMPurify.sanitize` to sanitize the `HTML` in my Vue apps. I've read that TipTap is removing unwanted html tags on its own but is this synonymous to sanitizing by a dedicated package like the one mentioned above?

**Comment by codemzy** [comment]
I think TipTap does a good job in the browser, since it only allows the tags (and attributes) that are specified in the extensions added to the editor instance. With that said, if you have extensions like links that allow urls to be entered, you might want to validate/sanitize that.

Always validate what is sent to the server though https://tiptap.dev/guide/output#security - I send the HTML output through [sanitize-html](https://www.npmjs.com/package/sanitize-html) on the server before storing it anywhere.

**Reply by sinclairnick** [reply]
Thanks for this input. I found the security docs a bit light so having someone's 2 cents is helpful here.

**Reply by tszemzo** [reply]
I gave a light reading sanitize-html docs and I found it very useful to filter or allow/disallow stuff from the output, but what would you typically validate (in my case I want to sanitize for security reasons)
